### PR TITLE
test(jetbrains): add archive extraction tests for AgnixBinaryDownloader

### LINKED ISSUE #255: JetBrains: add archive extraction tests for AgnixBinaryDownloader
## Summary
Add focused tests for JetBrains binary archive extraction paths in AgnixBinaryDownloader.

## Why
PR review flagged a coverage gap around TAR.GZ/ZIP extraction behavior and traversal protections.

## Scope
- Add TAR.GZ fixture extraction test
- Add ZIP fixture extraction test
- Assert target binary selection behavior
- Assert traversal protection and error behavior on malformed archives

## Context
Follow-up from PR #253 review thread (Copilot suggestion).

### PR DESCRIPTION
**Pull request by avifenesh**
## Summary
- Add 19 focused tests for archive extraction paths in `AgnixBinaryDownloader`
- Move 4 private methods (`extractTarGz`, `extractZip`, `isTargetBinaryEntry`, `verifyPathWithinDestination`) to companion object with `internal` visibility for testability
- Remove redundant `logger.error` calls from extract methods (caller already handles missing binary)

## Test Coverage
- **isTargetBinaryEntry** (6 tests): exact match, Unix/Windows subdirectories, rejection cases, case sensitivity
- **verifyPathWithinDestination** (3 tests): valid paths, path traversal rejection (SecurityException), boundary case
- **extractTarGz** (5 tests): root-level, subdirectory, missing binary, directory entry skipping, first-match behavior
- **extractZip** (5 tests): root-level, subdirectory, missing binary, directory entry skipping, first-match behavior

## Test Plan
- [x] All 24 JetBrains tests pass (`./gradlew test`)
- [x] Build passes (`./gradlew build -x test`)
- [x] No regressions to existing 5 URL trust tests

Closes #255

### CODE REVIEW

File: editors/jetbrains/src/main/kotlin/io/agnix/jetbrains/binary/AgnixBinaryDownloader.kt [hunk 1]
```diff
@@ -69,6 +69,76 @@ class AgnixBinaryDownloader {
...
+        /**
+         * Verify that an output file path is within the destination directory.
+         * Appends separator to prevent prefix matching issues (e.g., /tmp/agnix vs /tmp/agnix-other).
+         */
+        internal fun verifyPathWithinDestination(outFile: File, destination: File) {
+            val canonicalDest = destination.canonicalPath + File.separator
+            val canonicalOut = outFile.canonicalPath
+            if (!canonicalOut.startsWith(canonicalDest) && canonicalOut != destination.canonicalPath) {
+                throw SecurityException("Output path escapes destination directory: $canonicalOut")
+            }
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply editors/jetbrains/src/main/kotlin/io/agnix/jetbrains/binary/AgnixBinaryDownloader.kt]: Applied. Switched to Path.normalize() which handles lexical path resolution without I/O, sufficient for this defense-in-depth check since the output filename is already fixed.
